pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -1215,12 +1215,18 @@ private void createNewConsumingSegment(TableConfig tableConfig, PartitionLevelSt
     StreamPartitionMsgOffset partitionGroupSmallestOffset =
         getPartitionGroupSmallestOffset(streamConfig, partitionGroupId);
 
-    // Start offset must be higher than the start offset of the stream
-    if (partitionGroupSmallestOffset.compareTo(startOffset) > 0) {
-      LOGGER.error("Data lost from offset: {} to: {} for partition: {} of table: {}", startOffset,
-          partitionGroupSmallestOffset, partitionGroupId, tableConfig.getTableName());
+    if (partitionGroupSmallestOffset != null) {
+      // Start offset must be higher than the start offset of the stream
+      if (partitionGroupSmallestOffset.compareTo(startOffset) > 0) {
```

Comment:
assuming startOffset is never null yes?

Reply:
This is fine but what I am more concerned about is why it comes null in the first place. We definitely need this check but will need to dig a bit deeper as well for future PRs